examples/onnxruntime/optimization/text-classification/run_glue.py

```diff
@@ -130,12 +130,16 @@ class ModelArguments:
         default=None,
         metadata={"help": "Where do you want to store the pretrained models downloaded from huggingface.co"},
     )
+    ort_provider: str = field(
```

Comment:
@echarlaix What do you think of removing the argument in the examples and simply using

```python
    if optim_args.opt_level > 1 and optim_args.optimize_for_gpu:
        execution_provider = "CUDAExecutionProvider"
    else:
        execution_provider = "CPUExecutionProvider"
```
?

Reply:
I think it could make sense to let the user select the execution provider (to have the same behavior as in the quantization examples + the user might want to use `"CUDAExecutionProvider"` with optimization level of 1)

Reply:
I see, makes sense. In this case, you suggest to leave `ort_provider` as default `CPUExecutionProvider` and leave it as a parameter, while we overwrite `optim_args.optimize_for_gpu` to `True` if `optimization_level > 1`? Is this what you were suggesting above?

Reply:
My proposal is to let the user select the optimization parameters freely (and not overwrite anything) but to raise an error when those parameters are incompatible (in order to keep things as transparent and understandable as possible). What do you think ?

Reply:
Agreed, as you saw I added the following checks: https://github.com/fxmarty/optimum/blob/638e80e2bd8e86630162a964f98ae51d771fb407/examples/onnxruntime/optimization/text-classification/run_glue.py#L202-L222 . I hope it is not too heavy.

Reply:
It looks great, thanks for the addition !